### DISCUSSION TITLE
Does this work with static sites?

### DISCUSSION THREAD

**Discussion by rizkysyazuli** [LuXDAmore/nuxt-prune-html#35] [Q&A]
Just wanted to confirm if this works with static sites or not. Because the readme is a bit unclear. First, it says:

> If you `generate` your site it's not possibile to check request.headers, so [...] it always prune

So basically not gonna work right? Can't detect bots without it. But then, it also says:

> but You can disable this behavior by setting `hookGeneratePage` to `false`

I'm under the impression that this additional setting will make it work on static sites. Or am I wrong? Coz I'm still seeing all the scripts being loaded.

> or by using the type `query-parameters`

Not really sure about this. On a live site, sending bots to a different URL *is* considered cloaking.

Anyway, great work! Sorry for the long question.

**Comment by LuXDAmore** [comment]
Sorry for my late reply! @rizkysyazuli 

As written in the readme, setting `hookGeneratePage` to `true` will:
> If you generate your site it's not possible to check request.headers, so [...] it always prune
So, will always prune all the generate Javascript..
So your generated-HTML still valid, but without any Javascript inside (no spa navigation for example).

> **This option is useful only if you're site is fully static with zero-byte Javascript.**

This plugin is very useful in server-side-rendering-mode, but for static generation there is no server, no query-parameters and no headers, so there is no way to check if server has to send pure-HTML or the "normal-one".
